rpc/filters/filter.go

```diff
@@ -220,6 +220,11 @@ func (f *Filter) unindexedLogs(ctx context.Context, end uint64) ([]*types.Log, e
 	var logs []*types.Log
 
 	for ; f.begin <= int64(end); f.begin++ {
+		select {
+		case <-ctx.Done():
```

Comment:
so we didn't have timeout on these ranging rpcs?

Reply:
No. Even now we do not have timeout mechanism in this context. I would suggest to add a default timeout for all RPCs after Alex migrated Ethereum RPC module.